### DISCUSSION TITLE
Example for Speech To Text Live Transcription Using Node SDK

### DISCUSSION THREAD

**Discussion by Rajan-Gupta1** [deepgram/community#670] [General help]
Hello
Is there a an example similar to this https://github.com/deepgram-devs/browser-mic-streaming using deepgram SDK? I have tried various options but it is going into on.error() with following error. Not sure why this error is being returned by sdk
{
    "event": {
        "isTrusted": true
    },
    "message": "Unable to parse `data` as JSON.",
    "error": {}
}

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
We have the live-starter for Javascript: https://github.com/deepgram-starters/live-node-starter

This shows how to get audio from the browser microphone and stream it to Deepgram for real-time transcription.

**Comment by Rajan-Gupta1** [comment]
Thank You @SandraRodgers . I have studied this example and wanting to build the entire speech to text transcription on client side not requiring a server component other than to get a temp key. I have made it work with https://github.com/deepgram-devs/browser-mic-streaming but not able to make it work with DeepGram SDK. Any example using JS SDK which works on client?
Here is a sample of the code . I have tried putting getUserMedia outside too before createClient but 'the following is never called. When I use websockets instead of DeepGram SDK it works just fine. What could I be doing wrong or an example with using SDK on Client would be greatly helpful
 live.on(deepgram.LiveTranscriptionEvents.Transcript, (data) => {
        const received = JSON.parse(data);
        const transcript = received.channel.alternatives[0].transcript;
        console.log(transcript);
        if (transcript && received.is_final) {
          insertAtCursor(transcript);
        }
 });
async function initialize() {
  deepGramClient = deepgram.createClient(DG_KEY);
  live = deepGramClient.listen.live({
    model: "nova-2",
    language: "en-US",
    interm_results: true,
    smart_format: true,
  });
  
    live.on(deepgram.LiveTranscriptionEvents.Open, () => {
      live.on(deepgram.LiveTranscriptionEvents.Transcript, (data) => {
        const received = JSON.parse(data);
        const transcript = received.channel.alternatives[0].transcript;
        console.log(transcript);
        if (transcript && received.is_final) {
          insertAtCursor(transcript);
        }
      });
      live.on(deepgram.LiveTranscriptionEvents.Close, () => {
        console.log("Connection closed.");
      });
      live.on(deepgram.LiveTranscriptionEvents.SpeechStarted, (data) => {
        console.log(data);
      });
      live.on(deepgram.LiveTranscriptionEvents.Metadata, (data) => {
        console.log(data);
      });
      live.on(deepgram.LiveTranscriptionEvents.UtteranceEnd, (data) => {
        const received = JSON.parse(data);
        const transcript = received.channel.alternatives[0].transcript;
        console.log(transcript);
        if (transcript && received.is_final) {
          insertAtCursor(transcript);
        }
      });
      live.on(deepgram.LiveTranscriptionEvents.Error, (err) => {
        console.log(err);
      });
      navigator.mediaDevices.getUserMedia({
        audio: true,
        video: false
      }).then((stream) => {
        mediaRecorder = new MediaRecorder(stream, { mimeType: 'audio/webm' });
    
        mediaRecorder.addEventListener('dataavailable', event => {
          if (event.data.size > 0 && live.getReadyState() == 1) {
            live.send(event.data);
          }
        });
        mediaRecorder.start(100); // 250 milliseconds
      });
    }); // end of live.on
    startTimerKeepAlive(8000, live);
    stopTimerDeepGramFinish(60000, live, mediaRecorder);
}

**Comment by SandraRodgers** [comment]
Hi there, this example may help you. It's entirely client side. It imports the SDK from the CDN. Or do you still intend to bring in the SDK with npm?

https://github.com/deepgram-devs/js-livestream-example

The docs here show how to do this: https://developers.deepgram.com/docs/js-sdk-v2-to-v3-migration-guide#esm
